templates/start-dev-env.sh

```diff
@@ -0,0 +1,133 @@
+}
+
+function error_exit() {
+    echo "ERROR : $1"
+    exit 2
+}
+
+function can_i() {
+    command=$1
+    kubectl auth can-i $1 >& /dev/null || error_exit "No permission to $1"
+}
+
+# Start
+USER_NAME=$1
+DEV_PROJECT_ID=$2
+[[ -z "$USER_NAME" ]] && usage
+[[ -z "$DEV_PROJECT_ID" ]] && DEV_PROJECT_ID="001"
+
+echo '[Dev Environment]'
+
+# Validate cluster
+CLUSTER=$(kubectl config current-context)
+[[ ${CLUSTER} == "${PROJECT_NAME}-${ENVIRONMENT}-${REGION}" ]] || error_exit "Your kubernetes context ${CLUSTER} is not proper to run this script"
+echo "  Cluster: ${CLUSTER}"
+
+# Validate secret
+NAMESPACE=${PROJECT_NAME}
+SECRET_NAME=${PROJECT_NAME}
+DEV_SECRET_NAME=dev-${USER_NAME}
+DEV_SECRET_JSON=$(kubectl get secret ${DEV_SECRET_NAME} -n ${NAMESPACE} -o json)
+[[ -z "${DEV_SECRET_JSON}" ]] && error_exit "The secret ${DEV_SECRET_NAME} is not existing. Please check secrets with 'kubectl get secrets'"
+
+# Check installations
+if ! command_exist skaffold || ! command_exist telepresence; then
+    if ! command_exist skaffold; then
+        error_exit "command 'skaffold' not found: please visit https://skaffold.dev/docs/install/"
+    fi
+    if ! command_exist kubectl; then
+        error_exit "command 'telepresence' not found. You can download it at https://www.telepresence.io/reference/install."
+    fi
+fi
+
+# Setup dev namepsace
+DEV_USER_PROJECT=${USER_NAME}-${DEV_PROJECT_ID}
+DEV_NAMESPACE=${DEV_USER_PROJECT}
+kubectl get namespace ${DEV_NAMESPACE} >& /dev/null || \
+    (can_i "create namespace" && kubectl create namespace ${DEV_NAMESPACE} && \
+    can_i "create deployment -n ${DEV_NAMESPACE}" && \
+    can_i "create ingress -n ${DEV_NAMESPACE}" && \
+    can_i "create service -n ${DEV_NAMESPACE}" && \
+    can_i "create secret -n ${DEV_NAMESPACE}" && \
+    can_i "create configmap -n ${DEV_NAMESPACE}")
+echo "  Namespace: ${DEV_NAMESPACE}"
+
+# Setup dev secret from pre-configed one
+kubectl get secret ${SECRET_NAME} -n ${DEV_NAMESPACE} >& /dev/null || \
+    echo ${DEV_SECRET_JSON} | jq 'del(.metadata["namespace","creationTimestamp","resourceVersion","selfLink","uid"])' | sed "s/${DEV_SECRET_NAME}/${SECRET_NAME}/g" | kubectl apply -n ${DEV_NAMESPACE} -f -
+echo "  Secret: ${SECRET_NAME}"
+
+# Setup dev service account from pre-configured one
+SERVICE_ACCOUNT=backend-service
+kubectl get sa ${SERVICE_ACCOUNT} -n ${DEV_NAMESPACE} >& /dev/null || \
+    kubectl get sa ${SERVICE_ACCOUNT} -n ${NAMESPACE} -o json | jq 'del(.metadata["namespace","creationTimestamp","resourceVersion","selfLink","uid"])' | kubectl apply -n ${DEV_NAMESPACE} -f -
+
+# Setup dev k8s manifests, configuration, docker login etc
+CONFIG_ENVIRONMENT="staging"
+DEPLOYMENT=${PROJECT_NAME}
+REPO=${PROJECT_NAME}
+EXT_HOSTNAME=<% index .Params `stagingBackendSubdomain`  %><% index .Params `stagingHostRoot` %>
+MY_EXT_HOSTNAME=${DEV_USER_PROJECT}-${EXT_HOSTNAME}
+ECR_REPO=${ACCOUNT_ID}.dkr.ecr.${REGION}.amazonaws.com/${REPO}
+VERSION_TAG=lastest-${USER_NAME}
```

Comment:
typo - "lastest". Also, why would there be a tab called `latest-${USER_NAME}`?

Reply:
will not build and push user image to repo, instead, use existing Staging image. So this part of code will be dropped.